File: w01/d04/INSTRUCTORS/calculator/calculator.rb
class Calculator
  def add(x,y)
    x + y
  end

  def subtract(x,y)
    x - y
  end

  def multiply(x,y)
    x * y
  end

  def divide(x,y)
    return "NaN" if y.zero?
    x.to_f / y.to_f
  end
end

File: w06/d02/INSTRUCTORS/objects_in_js/calculator.js

var calculator = {
  add: function(a, b){
    return (a + b);
  },
  subtract: function(a, b){
    return (a - b);
  },
  multiply: function(a, b){
    return (a * b);
  },
  divide: function(a, b){
    return (a / b);
  },
  calc: function(a, b, operation){
    switch (operation){
      case 'add':
        return this.add(a, b);
        break;
      case 'subtract':
        return this.subtract(a, b);
        break;
      case 'multiply':
        return this.multiply(a, b);
        break;
      case 'divide':
        return this.divide(a, b);
        break;
    }
  }
}
